### ISSUE TITLE
`logit_bias` crashes natively, and `completion_probabilities` is never reset between completions (+ `n_probs` is post-sampling, unlike the docs)

### ISSUE THREAD

**Issue by kiyohken2000** [mybigday/llama.rn#401]
## Context

I'm using llama.rn for the "SemIf / OpenJev" pattern (https://openjev.com): semantic classification with a general-purpose GGUF by asking the model for exactly one option letter (`n_predict: 1`), restricting the output to the labels, and reading the label probabilities from `n_probs` instead of parsing generated text. SemIf's own worker does it with `max_tokens: 1`, `logprobs`, `logit_bias` on the label tokens and a `grammar` that only allows the labels; the probability lets the app threshold ("don't decide if unsure"). llama.rn has every ingredient for this and the automatic prefix reuse makes it cheap (one ~110-token prompt per decision, ~530 ms for Qwen3.5-2B on an A15), so it is a nice fit for on-device filtering — but three things get in the way:

1. `logit_bias` crashes the process (so the label bias step is unusable)
2. `completion_probabilities` returns stale data after the first call (so every decision after the first reads the wrong probabilities)
3. `n_probs` is post-sampling while the docs say raw softmax (so the readout is silently distorted unless every truncating sampler is disabled)

Reproduced on llama.rn 0.12.4 on an iPhone 13 mini (A15, iOS 27, Metal) with Qwen3.5-2B / Qwen3.5-0.8B / Qwen3-0.6B / gemma-3-1b-it Q4_K_M GGUFs; all three are still present in `main` (2af674a) by reading the source. 1 and 2 are plain bugs that affect any use of `logit_bias` / `ignore_eos` / `n_probs`, not just this pattern.

## 1. `logit_bias` (and `ignore_eos`) index a `std::vector` by token id → native crash

`cpp/jsi/JSIParams.cpp` (main, L581-600):

```cpp
sparams.logit_bias.clear();
...
if (ctx->params.sampling.ignore_eos) {
    sparams.logit_bias[llama_vocab_eos(vocab)].bias = -INFINITY;   // L586
}
...
sparams.logit_bias[tok].bias = val.get<double>();                  // L597
sparams.logit_bias[tok].bias = -INFINITY;                          // L599
```

`common_params_sampling::logit_bias` is `std::vector<llama_logit_bias>` (`vendor/llama.cpp/common/common.h` L278, element `{ llama_token token; float bias; }`), not the old `unordered_map<llama_token, float>`. After `clear()` the vector is empty, so `logit_bias[32]` is an out-of-bounds write. On device the process dies immediately.

Repro: any completion with `logit_bias: [[32, 100]]` (or `ignore_eos: true`) → crash before the first token. The same call without `logit_bias` works.

Fix is a one-liner per site: `sparams.logit_bias.push_back({ tok, bias });`.

## 2. `completion_probabilities` accumulates across completions (non-parallel path)

`cpp/rn-completion.cpp`: `generated_token_probs` is appended in `doCompletion()` (L1721) but never cleared — `rewind()` (L34) resets everything else but not this vector, and `JSICompletion.h` L156 serialises the whole vector into `completion_probabilities`. The parallel slot path does clear it (`cpp/rn-slot.cpp` L88), so only the classic `completion()` is affected.

Effect: on the 2nd, 3rd, … `completion()` on the same context with `n_probs > 0`, `completion_probabilities[0]` is still the first token of the *first* completion, and the array keeps growing. I first saw this as "every prompt returns exactly the same probabilities (to 15 digits) while `text` changes", on four different models.

Repro:

```ts
const a = await ctx.completion({ prompt: 'A or B?', n_predict: 1, n_probs: 5 })
const b = await ctx.completion({ prompt: 'C or D?', n_predict: 1, n_probs: 5 })
// b.completion_probabilities.length === 2, and [0] deep-equals a.completion_probabilities[0]
```

Fix: `generated_token_probs.clear();` in `rewind()`.

## 3. `n_probs` returns post-sampling probabilities; the doc comment says raw softmax

`src/types.ts` L287 says the probabilities are "a simple softmax of the logits without considering any other sampler settings", but `cpp/rn-completion.cpp` L1639 reads `common_sampler_get_candidates(ctx_sampling, true)` after `common_sampler_sample`, i.e. after the whole chain (top_k / top_p / min_p / temperature / grammar / …). Measured: with `grammar: 'root ::= "A" | "B"'` the top-2 come back as A 0.920 / B 0.080, without grammar A 0.916 / B 0.083 — the grammar renormalisation is visible in the numbers.

That is fine as a behaviour, but for probability readouts (SemIf-style thresholds, calibration, or any logprob consumer) it means you must set `temperature: 1, top_k: 0, top_p: 1, min_p: 0` yourself or the values are silently truncated/renormalised — with the defaults, `min_p: 0.05` alone zeroes the weaker label as soon as it drops below 5% of the stronger one. Proposal, mirroring llama.cpp server:

- add `post_sampling_probs?: boolean` (default `true` = current behaviour; `false` = softmax of raw logits from `llama_get_logits_ith`), and fix the comment on `n_probs`
- add `tok_id` next to `tok_str` in `NativeCompletionTokenProbItem` (the C++ side already has the id at L1639; matching labels by string is fragile across tokenizers, e.g. `"A"` vs `" A"`)

Happy to send PRs for 1 and 2 (both tiny), and for 3 if the shape above is acceptable.

**Comment by jhen0409** [comment]
Thanks! Released v0.13.0-rc.5, feel free to report or submit PRs if still have issue.
